test/parallel/test-fs-truncate.js

```diff
@@ -239,6 +239,41 @@ function testFtruncate(cb) {
   }));
 }
 
+{
+  const file7 = path.resolve(tmp, 'truncate-file-7.txt');
+  fs.writeFileSync(file7, 'Hi');
+  fs.truncate(file7, undefined, common.mustCall(function(err) {
+    assert.ifError(err);
+    assert(fs.readFileSync(file7).equals(Buffer.from('')));
+  }));
+}
+
+{
+  const file8 = path.resolve(tmp, 'truncate-file-8.txt');
```

Comment:
Nit: maybe use some more descriptive filename that will indicate that this file is indeed non-existent and make sure that nobody creates it accidentally (via other test), perhaps: `'non-existent-truncate-file.txt'`?